modelscope_agent/tools/modelscope_tools/pipeline_tool.py

```diff
@@ -0,0 +1,105 @@
+import os
+from typing import Dict, Optional
+
+import json
+import requests
+from modelscope_agent.tools.base import BaseTool
+from requests.exceptions import RequestException, Timeout
+
+MAX_RETRY_TIMES = 3
+
+
+class ModelscopePipelineTool(BaseTool):
+    default_model: str = ''
+    task: str = ''
+    model_revision = None
+
+    def __init__(self, cfg: Optional[Dict] = {}):
+        """
+        初始化一个ModelscopePipelineTool类
+        Initialize a ModelscopePipelineTool class.
+
+        Args:
+            cfg: (Dict[str, object]) 配置字典，包含了初始化对象所需要的参数
+        """
+        super().__init__(cfg)
+
+        self.api_url = self.cfg.get('url', self.url)
+        self.api_token = os.getenv('MODELSCOPE_API_TOKEN', None)
+        assert self.api_token is not None, 'api_token is not set'
+
+        # local call should be used by only cfg
+
+        self.use_local = not self.cfg.get('is_remote_tool', True)
+        self.is_initialized = False
+
+        if self.use_local:
+            self.model = self.cfg.get('model', None) or self.default_model
+            self.model_revision = self.cfg.get('model_revision',
+                                               None) or self.model_revision
+
+            self.pipeline_params = self.cfg.get('pipeline_params', {})
+            self.pipeline = None
+            self.is_initialized = False
+
+    def setup(self):
+        from modelscope.pipelines import pipeline
+
+        # only initialize when this tool is really called to save memory
+        if not self.is_initialized:
+            self.pipeline = pipeline(
+                task=self.task,
+                model=self.model,
+                model_revision=self.model_revision,
+                **self.pipeline_params)
+        self.is_initialized = True
+
+    def call(self, params: str, **kwargs) -> str:
```

Comment:
call should receive a arg called `remote` to do switch between local and remote call?

Reply:
yeah. I agree, maybe modelscope_hub_tool should implements the local call and modelscope_api_tool should implements the remote call